lib/Sema/TypeCheckAttr.cpp

```diff
@@ -3008,8 +3008,29 @@ void AttributeChecker::visitPropertyWrapperAttr(PropertyWrapperAttr *attr) {
 }
 
 void AttributeChecker::visitFunctionBuilderAttr(FunctionBuilderAttr *attr) {
-  // TODO: check that the type at least provides a `sequence` factory?
-  // Any other validation?
+  auto *nominal = dyn_cast<NominalTypeDecl>(D);
+  auto lookupResult = TypeChecker::lookupMember(
+      nominal, nominal->getDeclaredType(),
+      DeclNameRef(D->getASTContext().Id_buildBlock));
+
+  bool hasStaticBuildBlock = false;
+  llvm::SmallVector<ValueDecl *, 4> nonStaticBuildBlocks;
+  for (auto entry : lookupResult.innerResults()) {
+    auto *VD = entry.getValueDecl();
+    if (VD->getInterfaceType()->is<AnyFunctionType>()) {
+      if (VD->isStatic())
+        hasStaticBuildBlock = true;
+      else if (VD->getDeclContext() == nominal)
+        nonStaticBuildBlocks.push_back(VD);
```

Comment:
I think you can just do `if (!VD->isStatic()) { nonStaticBuildBlocks.pushBack(VD); }` and then later do `if (!nonStaticBuildBlocks.empty())` and diagnose each candidate.

Reply:
No, it probably makes sense to offer the fix-it for extension instance methods of the same type. The main goal here was to exclude `buildBlock` instance methods that come from e.g. protocol extensions.

Reply:
Reworked this in the latest commit so that we actually check which nominal type contains the non-`static` decl.